src/socket.c

```diff
@@ -1041,6 +1041,56 @@ void pthreads_socket_get_last_error(zval *object, zend_bool clear, zval *return_
 	}
 }
 
+void pthreads_socket_create_listen(zend_long port, zend_long backlog, zval *return_value) {
+	struct sockaddr_in  la;
+	struct hostent		*hp;
+	pthreads_object_t	*created;
+	pthreads_socket_t	*sock;
+
+	object_init_ex(return_value, pthreads_socket_entry);
```

Comment:
This can return `FAILURE`. It's not immediately clear to me if any such paths exists where this can fail (from a quick look through the source code), but it may be worthwhile checking its return value for failure just in case I've missed some psychotic edge case.